src/compatTest/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPluginTests.kt

```diff
@@ -885,6 +885,24 @@ class NexusPublishPluginTests {
         )
     }
 
+    @Test
+    fun `should find staging repository by description`() {
+        //given
+        writeDefaultSingleProjectConfiguration()
+        writeMockedSonatypeNexusPublishingConfiguration()
+        //and
+        val stagingRepository = StagingRepository(STAGED_REPOSITORY_ID, StagingRepository.State.OPEN, false)
+        val responseBody = getStagingReposWithOneStagingRepoWithGivenIdJsonResponseAsString(stagingRepository)
+        stubGetStagingReposForStagingProfileIdWithResponseStatusCodeAndResponseBody(STAGING_PROFILE_ID, 200, responseBody)
+
+        val result = run("findSonatypeStagingRepository")
```

Comment:
It fails. I suspect there is something with the regex.

>  No staging repository found for stagingProfileId: someProfileId, descriptionRegex: \Qorg.example:sample:0.0.1\E(\s|$) (found 1 in total)

is returned, even if Wiremock returns:
>               "description": "org.example:sample:0.0.1",

In general, I think that integration test should just the main scenario - the repo is found (and maybe a failure message). How would you see testing the regex itself (and surrounding logic) in more "unit" way?

Reply:
The way to test regex is to factor `firstOrNull` to a separate method, or even add `internal` method like `fun findStagingRepository(response: Response<Dto<List<ReadStagingRepository>>>, regex: Regex)`.
Then it could be tested without executing a call.

However, I do not see that to be a pattern yet: other tests are integration-only.

Reply:
The bug was `\b` instead of `\\b` 🤷

Reply:
Things happen. Regexes definitely can be tricky, so - even more - it  would be good to have it thoroughly tested. Gradle doesn't help ease unit testing, but `NexusClient` is not so Gradle bound. Maybe we could the second constructor in `NexusClient` accepting `NexusApi`? Then, in tests we could just inject there a `NexusApi` mock and stub `api.getStagingRepositories()`. WDYT?

Reply:
It might make sense to rework the approach to testing, however, it sounds like out of scope for the PR.

I would say a more important test here is to test stage and publish in independent gradle executions.

However, it was not clear if you would ever review/accept the feature, so I did not spent time on testing

Reply:
> It might make sense to rework the approach to testing, however, it sounds like out of scope for the PR.

The basic case with matching passes, so we could just create a separate issue to rework it and write more unit tests for that regexp (just to have it covered, assuming changes in the future).

> I would say a more important test here is to test stage and publish in independent gradle executions.

Good point. The parameter can be already passed from the command line in the second invocation, so it *should* behave here in the same way, however, definitely it would be good to test it :). I would prefer to do it with Wiremock (yes, I know that `NexusPublishPluginTests` is asking to be refactored, but let's sort the other things out first :-/).

> However, it was not clear if you would ever review/accept the feature, so I did not spent time on testing

AFAIR, you requested that feature to be used (among others) in the Apache Foundation projects where "staged" artifacts should be additionally tested/verified. That a valid use case, so - in general - I had been always open to accept it (especially that you provide the implementation :-) ). I just didn't precise which "August" I had in mind ;-).

Reply:
I've added a test to verify cases when the repository is not found